.travis.yml

```diff
@@ -2,12 +2,27 @@ language: c
 sudo: required
 env:
   - TARGET=local
-  - TARGET=docker
+  - TARGET=docker OS=ubuntu-16.04
+  - TARGET=docker OS=fedora-25
 services:
   - docker
 before_install:
-  - if [ ${TARGET}="local" ];  then sudo apt-get update && sudo apt-get install libsqlite3-dev; fi
-  - if [ ${TARGET}="docker" ]; then sudo docker build -t "cangjians/libcangjie:latest" . ; fi
+  - | # before_install: local
+    if [ "${TARGET}" = "local"  ]; then
+      sudo apt-get update && sudo apt-get install libsqlite3-dev
+    fi
+  - | # before_install: docker
+    if [ "${TARGET}" = "docker" ]; then
+      sudo docker pull "cangjians/build-essential:${OS}"
+      sudo docker tag "cangjians/build-essential:${OS}" "cangjians/build-essential:latest"
+      sudo docker build -t "cangjians/libcangjie" .
```

Comment:
Our images are based on pre-built Docker images (all different OS). But since the stock images don't naturally have the software and libraries for building `libcangjie`, `pycangjie` and `ibus-cangjie`, we have to further specify in our Dockerfile.